common/cards/card-plugins/effects/totem.js

```diff
@@ -9,30 +8,35 @@ class TotemEffectCard extends EffectCard {
 			name: 'Totem',
 			rarity: 'ultra_rare',
 			description:
-				'Player recovers +10hp after being knocked out and remains in battle.\n\nDiscard when applied.',
+				'Recover 10hp and remain in battle after you are knocked out.\nDoes not count as a knockout. Discard after use.',
 		})
-		this.recoverAmount = 10
 	}
 
 	/**
 	 * @param {GameModel} game
+	 * @param {string} instance
+	 * @param {import('../../../types/cards').CardPos} pos
 	 */
-	register(game) {
-		// attacks
-		game.hooks.attack.tap(this.id, (target) => {
-			if (target.effectCardId === this.id) {
-				target.recovery.push({amount: this.recoverAmount, discardEffect: true})
-			}
-			return target
-		})
+	onAttach(game, instance, pos) {
+		pos.otherPlayer.hooks.afterAttack[instance] = (result) => {
+			if (!result.attack.target.row || result.attack.target.row.health > 0)
```

Comment:
Please check explicitly that the row health is not undefined. You can also define a "row" variable which is result.attack.target.row, to keep the code cleaner.